File: 2020/Python/Day5.py
import os
import sys
import re


def main():
    inf = os.path.dirname(os.getcwd())
    inf = os.path.join(inf, "input")
    if not os.path.exists(inf) or os.path.isdir(inf):
        print(inf + " does not exist!")
        inf += ".txt"
        print(f"trying {inf} instead.")

    if not os.path.exists(inf) or os.path.isdir(inf):
        print(inf + " does not exist!")
        inf = os.path.join(os.path.dirname(inf), "input")
        inf = os.path.join(inf, "Day5.txt")
        print(f"trying {inf} instead.")

    if not os.path.exists(inf) or os.path.isdir(inf):
        print("None of the expected inputs exist!", file=sys.stderr)
        return
    else:
        print(f"Using input file {inf}.")

    inputFile = open(inf, 'r')
    found = []
    for line in inputFile.readlines():
        line = re.sub("[FL]", "0", re.sub("[BR]", "1", line))

        found.append(int(line, 2))

    found.sort()
    print(f"Highest found seat id is {found[-1]}.")

    last = 0
    for seat in found:
        if seat == last + 2:
            last = seat - 1
            break
        last = seat

    print(f"Your seat has id {last}.")

if __name__ == '__main__':
    main()

File: 2020/Java/src/aoc/Day5.java
package aoc;

import java.io.IOException;
import java.util.Arrays;
import java.util.List;

public abstract class Day5 {

	public static void main(String[] args) throws IOException {
		List<String> lines = InputReader.readInputFileLines(5);
		int[] found = new int[lines.size()];
		for (int i = 0; i < found.length; i++) {
			String line = lines.get(i).replaceAll("[FL]", "0").replaceAll("[RB]", "1");
			found[i] = Integer.parseInt(line, 2);
		}

		Arrays.sort(found);
		System.out.println("Highest found seat id is " + found[found.length - 1] + ".");

		int last = 0;
		for (int seat : found) {
			if (seat == last + 2) {
				last = seat - 1;
				break;
			}
			last = seat;
		}

		System.out.println("Your seat has id " + last + ".");
	}
}
